Nodejs/Product/Npm/SPI/NpmController.cs

```diff
@@ -15,23 +15,31 @@
 //*********************************************************//
 
 using System;
-using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.NodejsTools.Npm.SPI {
-    internal class NpmController : AbstractNpmLogSource, INpmController {
+    /// <summary>
+    /// Standard `INpmGlobalPackageProvider` that uses `npm bin` to get global packages.
+    /// </summary>
+    public class NpmBinGlobalPackageProvider : INpmGlobalPackageProvider {
+        public async Task<IGlobalPackages> GetGlobalPackages(string pathToNpm) {
+            var command = new NpmBinCommand(String.Empty, true, pathToNpm);
+            return (await command.ExecuteAsync())
+                ? RootPackageFactory.Create(command.BinDirectory)
+                : null;
+        }
+    }
 
+    internal class NpmController : AbstractNpmLogSource, INpmController {
         private IPackageCatalog _sRepoCatalog;
         private string _fullPathToRootPackageDirectory;
         private string _cachePath;
         private bool _showMissingDevOptionalSubPackages;
         private INpmPathProvider _npmPathProvider;
-        private IRootPackage _rootPackage;
-        private IGlobalPackages _globalPackage;
-        private readonly object _lock = new object();
```

Comment:
Will revert this. I believe auto properties are threadsafe in that reference get/set are atomic so you can't end up reading a bad pointer, but not threadsafe in that they do not force synchronization of threads, so another thread may end up reading a previous cached value instead of the most recently set one. I'm not convinced we need these locks (and there's a ton of other places that should use them if we did) but keeping them does not hurt anything.